### PR TITLE
Add sample step handler

### PR DESCRIPTION
**Pull request by emswbandara**
## Purpose
Sample implementation of a step handler for the purpose of orchestrating invocation of authenticators based on the availability of the IWAKerberosAuthenticator in the authenticator list.

### CODE REVIEW

File: etc/sample-step-handler/src/main/java/org/wso2/carbon/identity/handler/step/CustomStepHandler.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+                    locatedIWAAuthenticatorConfig = config;
+                } else if (CustomStepHandlerConstants.BASIC_AUTHENTICATOR_NAME.equals(config.getName())) {
+                    locatedBasicAuthenticatorConfig = config;
+                }
+            }
+        } else {
+            if (log.isDebugEnabled()) {
+                String spName = context.getServiceProviderName();
+                int currentStep = context.getCurrentStep();
+                log.debug("authConfigList is empty in the step : " + currentStep + " of service provider : " + spName);
```
Comment (vihanga-liyanage) [inline etc/sample-step-handler/src/main/java/org/wso2/carbon/identity/handler/step/CustomStepHandler.java]: Let's follow the 120 characters for a line rule.

File: etc/sample-step-handler/src/main/java/org/wso2/carbon/identity/handler/step/CustomStepHandler.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        boolean isInternalIP = CustomStepHandlerUtil.isInternalIP(clientIpAddress);
+
+        /*
+         * modify the authConfigList based on the IP address to include IWA for internal network users and basic auth
+         * for external users.
+         */
+        if (isInternalIP) {
+            if (locatedIWAAuthenticatorConfig != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Client IP address is from internal range. Hence modifying the authConfigList to include " +
```
Comment (vihanga-liyanage) [inline etc/sample-step-handler/src/main/java/org/wso2/carbon/identity/handler/step/CustomStepHandler.java]: Please make the character count less than 120 per line.

File: etc/sample-step-handler/src/main/java/org/wso2/carbon/identity/handler/step/utils/IPRangeDataHolder.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    /**
+     * Read internal IP ranges from the config file 'internal_ip_addresses.json' and populate to a list.
+     *
+     * @return
+     */
+    private void readIPListFromConfig() {
+
+        if (Files.exists(Paths.get(CustomStepHandlerConstants.IP_CONFIG_FILE_PATH))) {
+            StringBuilder stringBuilder = new StringBuilder();
+            try (BufferedReader in = new BufferedReader(new FileReader(CustomStepHandlerConstants.IP_CONFIG_FILE_PATH))) {
```
Comment (vihanga-liyanage) [inline etc/sample-step-handler/src/main/java/org/wso2/carbon/identity/handler/step/utils/IPRangeDataHolder.java]: 120 characters per line.

File: etc/sample-step-handler/pom.xml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            <artifactId>org.wso2.carbon.identity.application.authentication.framework</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.felix</groupId>
+            <artifactId>org.apache.felix.scr.ds-annotations</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.googlecode.json-simple</groupId>
+            <artifactId>json-simple</artifactId>
+            <version>1.1.1</version>
```
Comment (Kanapriya) [inline etc/sample-step-handler/pom.xml]: The versions can be move to the properties file

File: etc/sample-step-handler/src/main/java/org/wso2/carbon/identity/handler/step/CustomStepHandler.java [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+                }
+            }
+        } catch (Exception e) {
+            log.error("Error occurred during executing custom step handler.", e);
+        }
+
+        super.handle(request, response, context);
+
+    }
+
```
Comment (Kanapriya) [inline etc/sample-step-handler/src/main/java/org/wso2/carbon/identity/handler/step/CustomStepHandler.java]: Let's have some method level comments?

**Comment by madurangasiriwardena** [comment]
@emswbandara can you please attend to the comments added?

**Pull request merged**